### PR TITLE
Add hover state to `TableColumnHeader` when 'allowSorting' is true

### LINKED ISSUE #5284: Lack of `data-hovered`  on Table column headers
### Provide a general summary of the feature here

Hi, I'm not sure whether this is a bug or feature request, but I'm not seeing `data-hovered` added to column headers, especially if they're sortable, which would point to the user about possible interactivity possibility.

Not sure if related to https://github.com/adobe/react-spectrum/issues/4411

### 🤔 Expected Behavior?

Ability to style Table Column headers using RAC patterns, with `data-hovered` selector.

### 😯 Current Behavior

Need to reach for `:hover` selector

### 💁 Possible Solution

_No response_

### 🔦 Context

I have a use-case where I'm using sortable column headers. To make it more clear to the user about interactivity behind it, I'd like to apply additional styles once the column header is hovered.

### 💻 Examples

Any RAC table example – by looking at devtools we can see that `data-hovered` is not added.

### 🧢 Your Company/Team

_No response_

### 🕷 Tracking Issue

_No response_

### PR DESCRIPTION
**Pull request by sookmax**
Closes #5284

Hello, this PR adds support for hover state (`data-hovered`) in `TableColumnHeader` when `Column` is sortable (`allowsSorting` = `true`). 

I'm not 100% sure whether `data-hovered` should only be supported when a column is sortable, but judging by the discussions in #5284 and #4411, it seems we want hover state support only when an item (whether a column or a row) is interactive? I'd love to know you guys' thoughts on this.

Thanks!

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by snowystinger** [comment]
GET_BUILD

**Review by snowystinger — DISMISSED** [review]
We have plans for some other interactions, but I think they'll all be through a button of some variety. We have a few examples of this with resizable columns. I think Sort is the only one which results in a direct action. There may be other cases later on, but I think we can add them later. I think this is fine for now.

Verified data-hovered is applied in docs sorting example.

Thank you!

**Review by yihuiliao — APPROVED** [review]
lgtm! thanks for the pr!

File: packages/react-aria-components/src/Table.tsx [hunk 1]
```diff
@@ -804,7 +803,8 @@ function TableColumnHeader<T>({column}: {column: GridNode<T>}) {
     }
   }
 
-  let props: ColumnProps = column.props;
+  let props: ColumnProps = column.props
```
Comment (yihuiliao) [inline packages/react-aria-components/src/Table.tsx:806]:
```suggestion
  let props: ColumnProps = column.props;
```

**Review by yihuiliao — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

**Pull request merged**